fog/distribution/src/main.rs

```diff
@@ -355,23 +355,28 @@ fn worker_thread_entry(
 
         // Send to the next fog account
         let to_account = &fog_accounts[txs_created % fog_accounts.len()];
+        // Sometime transactions could not be submitted before tombstone block passed,
+        // so loop until transactions can be submmitted
+        let mut done = false;
+        while !done {
+            // Got our inputs, construct transaction.
+            let tx = build_tx(
+                &pending_spendable_txouts,
+                to_account,
+                &config,
+                &ledger_db,
+                fog_resolver.clone(),
+                &logger,
+            );
 
-        // Got our inputs, construct transaction.
-        let tx = build_tx(
-            &pending_spendable_txouts,
-            to_account,
-            &config,
-            &ledger_db,
-            fog_resolver.clone(),
-            &logger,
-        );
-
-        txs_created += 1;
+            txs_created += 1;
```

Comment:
the main point of this counter is that it determines which account is going to recieve the payment

we generally want to make sure all the accounts get funded for testing purposes

so when we rebuild, we don't want to rebuild the transaction for a new recipient

so we should only increment this counter after submit_tx has succeeded